src/browser/api_protocol/api_handlers/authorization.js

```diff
@@ -211,6 +211,26 @@ function AuthorizationApiHandler() {
 
     /*jshint unused:false */
     apiProtocolBase.registerActionMap(authenticationApiMap);
-}
+};
+
+const isConnectionAuthenticated = (msg, next) => {
+    const { data, nack, identity, strategyName } = msg;
+    const { uuid } = identity;
+    const action = data && data.action;
+
+    // Prevent all API calls from unauthenticated external connections,
+    // except for authentication APIs
+    if (
```

Comment:
@HarsimranSingh after taking a look at this again, after Ricardo pointed it out, one of the calls here is an expensive one, and we have to give up clarity here a bit